src/core/main.c

```diff
@@ -743,8 +744,17 @@ static void manager_set_defaults(Manager *m) {
         m->default_tasks_accounting = arg_default_tasks_accounting;
         m->default_tasks_max = arg_default_tasks_max;
 
-        manager_set_default_rlimits(m, arg_default_rlimit);
-        manager_environment_add(m, NULL, arg_default_environment);
+        r = manager_set_default_rlimits(m, arg_default_rlimit);
+
+        if (r)
+                return r;
```

Comment:
Hm, why? You don't provide any justification in the patch... Rlimits are usually not that important, and failing to start the manager increases the likelihood of failure. We tend to do that only for the most important issues, like SELinux failing to load or failing to start the default target.

As for the coding style: please don't leave an empty line between `r = ...` and `if (r)`, and also use `if (r < 0)`.